### DISCUSSION TITLE
Is this an inference only library?

### DISCUSSION THREAD

**Discussion by siddhsql** [ggml-org/llama.cpp#1855] [Q&A] [answered]
is `llama.cpp` an inference only library? how can i fine-tune an existing model using `llama.cpp`?

**Comment by Green-Sky** [comment, marked as answer]
training/finetuning is still experimental
latest update on that: https://github.com/ggerganov/llama.cpp/pull/1652

**Reply by siddhsql** [reply]
thanks
